cmd/deploy/deploy.go

```diff
@@ -320,30 +326,48 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
 
 	err = dc.deploy(ctx, deployOptions)
 
+	oktetoLog.SetStage("")
+	oktetoLog.DisableMasking()
+
 	if err != nil {
-		oktetoLog.AddToBuffer(oktetoLog.InfoLevel, "Deployment failed: %s", err.Error())
+		err = oktetoErrors.UserError{
+			E:    fmt.Errorf(oktetoErrors.ErrDeployHasFailedCommand, err),
+			Hint: "Please try updating your deploy section on your manifest and try again",
+		}
+		oktetoLog.AddToBuffer(oktetoLog.InfoLevel, err.Error())
 		data.Status = pipeline.ErrorStatus
 	} else {
-		oktetoLog.AddToBuffer(oktetoLog.InfoLevel, "'%s' successfully deployed", deployOptions.Name)
-		data.Status = pipeline.DeployedStatus
+		hasDeployed, err := pipeline.HasDeployedSomething(ctx, deployOptions.Name, deployOptions.Manifest.Namespace, c)
+		if err != nil {
+			return err
+		}
+		if hasDeployed {
+			if !utils.LoadBoolean(model.OktetoWithinDeployCommandContextEnvVar) {
+				if err := dc.showEndpoints(ctx, deployOptions); err != nil {
+					oktetoLog.Infof("could not retrieve endpoints: %s", err)
+				}
+			}
+			if deployOptions.ShowCTA {
+				oktetoLog.Success(succesfullyDeployedmsg, deployOptions.Name)
+				if oktetoLog.IsInteractive() {
+					oktetoLog.Green("    Run `okteto up` to activate your development container.")
+				}
+			}
+			data.Status = pipeline.DeployedStatus
+		} else {
+			err = oktetoErrors.UserError{
+				E:    oktetoErrors.ErrDeployHasNotDeployAnyResource,
+				Hint: "Please try updating your deploy section on your manifest and try again",
+			}
+			oktetoLog.AddToBuffer(oktetoLog.InfoLevel, err.Error())
+			data.Status = pipeline.ErrorStatus
```

Comment:
I think It is odd that we mark as error the deploy if nothing was deployed, specially in the UI. The deploy would appear as Error but the logs wouldn't show any error (as we don't add an stage here, this log wouldn't appear in the UI with the structured logs).

I don't know if there would make sense any scenario in which you execute a deploy without deploying anything, but depending on how other dependencies are deployed, there could be some scenarios.

If each dependency is deployed as an independent pipeline and I'm using my manifest just to deploy all the dependencies but it doesn't deploy anything by itself, I think this would happen